src/rdkafka_cgrp.c

```diff
@@ -862,20 +935,24 @@ rd_kafka_rebalance_op_incr (rd_kafka_cgrp_t *rkcg,
                              rd_kafka_error_string(error));
                 rd_kafka_error_destroy(error);
 
-                /* FIXME: Math; do you think this error handling is appropriate?
-                 * This will at least clean up the current mess. */
-                rd_kafka_assignment_set_lost(rkcg,
-                                             "internal (un)assign failed");
                 rd_kafka_cgrp_set_join_state(rkcg,
                                              /* This is a clean state for
                                               * assignment_done() to rejoin
                                               * from. */
                                              RD_KAFKA_CGRP_JOIN_STATE_STEADY);
-                rd_kafka_assignment_clear(rkcg);
+                rd_kafka_assignment_clear(rkcg->rkcg_rk);
         }
 
         /* Now serve the assignment to make updates */
-        rd_kafka_assignment_serve(rkcg);
+        rd_kafka_assignment_serve(rkcg->rkcg_rk);
```

Comment:
by inside out / upside down, i mean program structure which is strictly hierarchical is the easiest to understand (even if code required is more verbose). here we have rkcg referencing rk.

(not proposing changing now).